### DISCUSSION TITLE
Batch split and save RGB channels

### DISCUSSION THREAD

**Discussion by fizion** [ImageMagick/ImageMagick#5081] [Help]
While old I'm still better with graphics than with command lines and can't get the logic for file naming. Guess it's easy if you understand the logic but well, my brain is formatted for pictures.

I have a directory with DDS files which are packed files. The RGB channels are basically 3 separate grayscale images where R = Ambient Occlusion,  G = Roughness and B = Metallic.

I figured out the splitting is done by -separate, and if I run convert `*.dds -separate %d.png` it will separate all channels and gives the PNG results a number. Which is not that handy.

If a file in the directory is called 'livery_comp.dds' I would like the separate channels called something like 'livery_comp_ao', 'livery_comp_roughness' and 'livery_comp_metallic. So I thought maybe it needed some kind of loop since there are always 3 channels and it gets saved in the R-G-B order.

Whatever I tried using %channels %basename etc, it got really messy. I'm now thinking of programming a PHP thing on a local server since that feels more comfortable/logic to my brain than the CMD-box. But I though to try to ask first, googled my *ss off since I thought there had to be a certain tool specialized in splitting channels since it's so common for PBR/texture/3D work but no luck.

**Comment by snibgo** [comment]
What version of IM, on what platform? I will assume IM v7 on Windows BAT.

To save channels as different grayscale names, I suggest:
```
magick ^
  in.png ^
  -channel RGB -separate +channel ^
  ( -clone 0 -write out0.png +delete ) ^
  ( -clone 1 -write out1.png +delete ) ^
  ( -clone 2 -write out2.png +delete ) ^
  NULL:
```
Use your own filenames, of course.

If you have many input files, you can put this in  "for" loop.

**Reply by fizion** [reply]
Your assumptions are right, it's IMv7 on Windows. I just looked at creating loops and using variable names within that loop but I think the example I saw was for *nix.

```
magick ^
for f in *.dds
do
	-channel RGB -separate +channel ^
	( -clone 0 -write "${f%_ao.png}" +delete ) ^
	( -clone 1 -write "${f%_roughness.png}" +delete ) ^
	( -clone 2 -write "${f%_metallic.png}" +delete ) ^
done
NULL: 
```

Of course this doesn't work but shows what I want, loop through a bunch of DDS files and use a handy naming convention. I looked for FOR-loop under DOS but just can't get it to work.

**Comment by snibgo** [comment]
"NULL:" is part of the "magick" command, so should be inside the loop, not after it.

For help on Windows "for", type "help for" at the command line.
